Cbc/src/CbcCutGenerator.cpp

```diff
@@ -309,9 +309,20 @@ bool CbcCutGenerator::generateCuts(OsiCuts &cs, int fullScan, OsiSolverInterface
       // Pass across model information in case it could be useful
       //void * saveData = solver->getApplicationData();
       //solver->setApplicationData(model_);
-      generator_->generateCuts(*solver, cs, info);
+      const bool origModel = generator_->needsOriginalModel();
+      const OsiSolverInterface* solverNow = origModel ?
+            model_->originalSolver(0) : solver;           // very inefficient, NOT THREAD-SAFE
```

Comment:
This doesn't sound like something that one would want to have in master.

Reply:
This only affects combinatorial cuts and not any existing applications